src/widgets/concept-widget.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  VDomModel, VDomRenderer
+} from '@jupyterlab/apputils';
+import {INotebookTracker} from '@jupyterlab/notebook';
+
+import * as React from 'react';
+
+import {AllOfUsConfig} from '../config';
+import {Concept} from '../generated';
+import {ConceptsService} from '../services/concepts.service';
+
+import {Observable} from 'rxjs/Rx';
+
+export enum LoadingStatus {
+  UNSET,
+  LOADING,
+  LOADED
+}
+
+export class ConceptsWidgetModel extends VDomModel {
+
+  public initialized: boolean;
+  public loadingStatus = LoadingStatus.UNSET;
+  public query = '';
+  public concepts: Array<Concept>;
+  public selectedConcept: Concept;
+}
+
+export class ConceptsWidgetRenderer extends VDomRenderer<ConceptsWidgetModel> {
+
+
+  constructor(configObservable: Observable<AllOfUsConfig>,
+              private conceptsService: ConceptsService,
+              private notebooks: INotebookTracker) {
+    super();
+
+    this.id = 'allofus-concepts';
+    this.title.label = 'Concepts';
+    this.title.closable = true;
+    configObservable.subscribe((config) => {
+      if (!this.model.initialized) {
+        this.model.initialized = true;
+        this.update();
+      }
+    });
+  }
+
+  protected render(): React.ReactElement<any> {
+    const handleQueryChange = (event) => {
+      this.model.query = event.target.value;
+      this.update();
+    };
+
+    const searchConcepts = () => {
+      this.model.loadingStatus = LoadingStatus.LOADING;
+      this.update();
+      this.conceptsService.searchConcepts({query: this.model.query})
+          .then((concepts) => {
+            this.model.concepts = concepts;
+            this.model.loadingStatus = LoadingStatus.LOADED;
+            this.update();
+          });
+    };
+    return (
+        <div>
+          <input value={this.model.query} onChange={handleQueryChange}
+                 disabled={!this.model.initialized}/>
+          <br/>
+          <input value='Search' type='button' onClick={searchConcepts}
+                 disabled={!this.model.initialized || !this.model.query}/>
+          <br/>
+          {this.loadingImage()}
+          {this.conceptsTable()}
+        </div>
+    );
+  }
+
+  protected loadingImage(): React.ReactElement<any> {
+    if (this.model.loadingStatus !== LoadingStatus.LOADING) {
+      return null;
+    }
+    // Three divs inside the inner div makes three balls in the spinner.
+    // See: https://github.com/ConnorAtherton/loaders.css
+    return (
+        <div className='loaderWrapper'>
+          <div className='loader-inner ball-pulse'>
+            <div></div>
+            <div></div>
+            <div></div>
+          </div>
+        </div>
+    );
+  }
+
+  protected conceptsTable(): React.ReactElement<any> {
```

Comment:
I would suggest this be a class component on its own, e.g.`ConceptTable`. Further, I would break out the rows into another class as well. Right now, we're not doing much with the individual row content aside from loading the concept id, but one could imagine adding sorting and filtering at the table level.